src/Stratis.Bitcoin.Features.Interop/EthereumClient/EthereumClientBase.cs

```diff
@@ -0,0 +1,135 @@
+﻿using System.Collections.Generic;
+using System.Numerics;
+using Nethereum.ABI;
+using Nethereum.Contracts;
+using Nethereum.Hex.HexConvertors.Extensions;
+using Nethereum.Hex.HexTypes;
+using Nethereum.JsonRpc.Client;
+using Nethereum.RPC.Eth.Blocks;
+using Nethereum.RPC.Eth.DTOs;
+using Nethereum.Web3;
+using Nethereum.Web3.Accounts.Managed;
+using Stratis.Bitcoin.Features.Interop.Models;
+
+namespace Stratis.Bitcoin.Features.Interop.EthereumClient
+{
+    public class EthereumClientBase : IEthereumClientBase
+    {
+        private readonly InteropSettings interopSettings;
+        private readonly Web3 web3;
+        private Event<TransferEventDTO> transferEventHandler;
+        private NewFilterInput filterAllTransferEventsForContract;
+        private HexBigInteger filterId;
+
+        public const string ZeroAddress = "0x0000000000000000000000000000000000000000";
+
+        public EthereumClientBase(InteropSettings interopSettings)
+        {
+            this.interopSettings = interopSettings;
+
+            if (!this.interopSettings.Enabled)
+                return;
+
+            var account = new ManagedAccount(interopSettings.EthereumAccount, interopSettings.EthereumPassphrase);
+            
+            // TODO: Support loading offline accounts from keystore JSON directly?
+            if (!string.IsNullOrWhiteSpace(interopSettings.EthereumClientUrl))
+                this.web3 = new Web3(account, interopSettings.EthereumClientUrl);
+            else
+                this.web3 = new Web3(account);
+        }
+
+        public void CreateTransferEventFilter()
+        {
+            this.transferEventHandler = this.web3.Eth.GetEvent<TransferEventDTO>(this.interopSettings.WrappedStraxAddress);
+            this.filterAllTransferEventsForContract = this.transferEventHandler.CreateFilterInput();
+            this.filterId = this.transferEventHandler.CreateFilterAsync(this.filterAllTransferEventsForContract).GetAwaiter().GetResult();
+        }
+
+        public List<EventLog<TransferEventDTO>> GetTransferEventsForWrappedStrax()
```

Comment:
this method and the one that calls it can be made async to avoid doing `.GetAwaiter().GetResult();` in the code

Reply:
@noescape00 I suspect this is because of NEthereum's interface contract not being async or returning `Task`? @zeptin can you confirm?